packages/app/src/app/pages/Sandbox/Editor/Workspace/WorkspaceItem/index.tsx

```diff
@@ -17,68 +19,48 @@ type Props = {
   defaultOpen?: boolean;
   actions?: React.Component<any, any>;
   style?: React.CSSProperties;
-  showOverflow?: boolean;
 };
 
-type State = {
-  open: boolean;
-};
-
-export default class WorkspaceItem extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      open: Boolean(props.defaultOpen),
-    };
-  }
-
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
-    return (
-      nextState.open !== this.state.open ||
-      nextProps.disabled !== this.props.disabled ||
-      this.props.children !== nextProps.children
-    );
-  }
-
-  toggleOpen = () => this.setState({ open: !this.state.open });
+const WorkspaceItem = ({
+  children,
+  defaultOpen,
+  title,
+  keepState,
+  disabled,
+  actions,
+  style,
+}: Props) => {
+  const [open, setOpen] = useState(Boolean(defaultOpen));
+  const hidden = useHiddenState({ visible: Boolean(defaultOpen) });
 
-  render() {
-    const {
-      children,
-      title,
-      keepState,
-      disabled,
-      actions,
-      style,
-      showOverflow,
-    } = this.props;
-    const { open } = this.state;
-
-    return (
-      <>
-        <ItemHeader style={style} onClick={this.toggleOpen}>
+  return (
+    <>
+      <HideButton {...hidden}>
+        <ItemHeader style={style} onClick={() => setOpen(!open)}>
```

Comment:
```suggestion
        <ItemHeader style={style} onClick={hidden.toggle}>
```

This can be simplified to use hidden's internal state and methods only